templates/sample-app/.config.js

```diff
@@ -0,0 +1,232 @@
+	}
+}
+exports.name = yes ? name : prompt('package name', niceName(name), function (data) {
+	var its = validateName(data)
+	if (its.validForNewPackages) return data
+	var errors = (its.errors || []).concat(its.warnings || [])
+	var er = new Error('Sorry, ' + errors.join(' and ') + '.')
+	er.notValid = true
+	return er
+})
+
+var version = package.version || conf('init.version') || '1.0.0'
+exports.version = yes ?
+	version :
+	prompt('version', version, function (version) {
+		if (semver.valid(version)) return version
+		var er = new Error('Invalid version: "' + version + '"')
+		er.notValid = true
+		return er
+	})
+
+if (!package.description) {
+	exports.description = yes ? '' : prompt('description')
+}
+
+if (!package.bin) {
+	exports.bin = function (cb) {
+		fs.readdir(path.resolve(dirname, 'bin'), function (er, d) {
+			// no bins
+			if (er) return cb()
+			// just take the first js file we find there, or nada
+			return cb(null, d.filter(function (f) {
+				return f.match(/\.js$/)
+			})[0])
+		})
+	}
+}
+
+exports.directories = function (cb) {
+	fs.readdir(dirname, function (er, dirs) {
+		if (er) return cb(er)
+		var res = {}
+		dirs.forEach(function (d) {
+			switch (d) {
+				case 'example':
+				case 'examples':
+					return res.example = d
+				case 'test':
+				case 'tests':
+					return res.test = d
+				case 'doc':
+				case 'docs':
+					return res.doc = d
+				case 'man':
+					return res.man = d
+				case 'lib':
+					return res.lib = d
+			}
+		})
+		if (Object.keys(res).length === 0) res = undefined
+		return cb(null, res)
+	})
+}
+
+exports.dependencies = {
+	"@financialforcedev/orizuru": "^5.0.2",
+	"@financialforcedev/orizuru-auth": "^3.0.3",
+	"@financialforcedev/orizuru-transport-rabbitmq": "^3.0.3",
+	"@financialforcedev/orizuru-openapi": "^2.0.1",
+	"debug-plus": "1.2.2",
+	"fs-extra": "4.0.2",
+	"jsforce": "1.8.0",
+	"klaw-sync": "3.0.0",
+	"lodash": "4.17.4",
+	"pkginfo": "0.4.1",
+	"throng": "4.0.0",
+	"uuid": "3.1.0"
+}
+
+exports.devDependencies = {
+	"@financialforcedev/eslint-config": "^3.0.0",
+	"chai": "4.1.1",
+	"chai-as-promised": "7.1.1",
+	"mocha": "3.5.0",
+	"nyc": "11.2.1",
+	"proxyquire": "1.8.0",
+	"sinon": "3.2.1",
+	"sinon-chai": "2.14.0"
+}
+
+exports.scripts = {
+	"generate-apex-transport": "orizuru setup generate-apex-transport src/node/lib/schema/api src/apex/app/main/default/classes",
+	"lint-fix": "eslint src/node --fix",
+	"orizuru-post-init": "mvn clean install",
+	"pretest": "eslint src/node",
+	"single-test": "nyc --all=false mocha",
+	"test": "nyc mocha --recursive src/node/spec"
+}
+
+exports.nyc = {
+	"check-coverage": true,
+	"per-file": true,
+	"lines": 100,
+	"statements": 100,
+	"functions": 100,
+	"branches": 100,
+	"include": [
+		"src/node"
+	],
+	"reporter": [
+		"lcov",
+		"html",
+		"text",
+		"text-summary"
+	],
+	"cache": true,
+	"all": true
+}
+
+if (!package.repository) {
+	exports.repository = function (cb) {
+		fs.readFile('.git/config', 'utf8', function (er, gconf) {
```

Comment:
Not sure it needs to be quite so clever... each neat time-saver is admittedly neat, but I worry it adds a potential point of failure, e.g. if the user isn't using git for source control.
I'd say the user should just enter their repo details manually, and/or we put a massive try/catch around the whole nice-to-have functionality.
(Something for the backlog).